### ISSUE TITLE
Format the frontend code

### ISSUE THREAD

**Issue by Lauritz-Tieste** [Bootstrap-Academy/Bootstrap-Academy#134]
### Description

We need to ensure that our frontend codebase is consistently formatted to improve readability and maintainability.
To achieve this, we should implement Prettier, a popular code formatter, and integrate it with GitHub Checks.

**Comment by iFrozenPhoenix** [comment]
Hi, I would be interested in resolving this issue.

**Comment by Defelo** [comment]
At least for the frontend repo there is already a draft PR for this at https://github.com/Bootstrap-Academy/frontend/pull/284. @Lauritz-Tieste Are you still working on this?

**Comment by Lauritz-Tieste** [comment]
No, not really

**Comment by PeachpitRipley** [comment]
Hello, I would like to take over https://github.com/Bootstrap-Academy/Bootstrap-Academy/issues/134. Would it be possible to assign this task to me? Thank you

**Comment by PeachpitRipley** [comment]
Suggestion for Issue #134

While running npm run format:check I noticed that Prettier fails on a few Vue files due to invalid <p> nesting (1. components/calendar/EventSummary.vue; 2. components/form/CodingChallenge.vue; 3. pages/matchings/solve-[id].vue, 4. pages/quizzes/solve-[id].vue).

Since these are definite markup errors and currently block Prettier from running successfully, I suggest fixing them directly in this branch as part of #134. The changes would only be minor (replace nested `<p>` with `<span>` or remove extra closing tags).

Would you be fine with me including these fixes in this PR, or should we open a separate ticket for them?

**Comment by Defelo** [comment]
> Would you be fine with me including these fixes in this PR?

Yes, just make sure to split these changes into separate commits. i.e. first have one small commit to fix the markup errors and then one large, automatically generated (and reproducible!) commit to fix the formatting.

**Comment by TheMorpheus407** [comment]
Closed via https://github.com/Bootstrap-Academy/frontend/pull/467
